src/services/inference/module/module_js.ml

```diff
@@ -407,7 +407,14 @@ module Node = struct
     lazy_seq [
       lazy (
         if SSet.mem dir node_modules_containers then
-          lazy_seq (Files.node_resolver_dirnames file_options |> Core_list.map ~f:(fun dirname ->
+          lazy_seq (Files.node_resolver_aliases file_options |> Core_list.map (fun dirname ->
+            lazy (resolve_relative
+              ~options
+              loc ?resolution_acc dir (spf "%s%s%s" dirname Filename.dir_sep r)
+            )
+          ))
+        else if SSet.mem dir node_modules_containers then
```

Comment:
This doesn't look right to me. This is checking the same exact thing as the `if` above, so this code path will never be traversed. Am I missing something?

Reply:
Modified the handling here but absolutely needs a review :-D